langchain/vectorstores/pinecone.py

```diff
@@ -29,13 +33,13 @@ class Pinecone(VectorStore):
             pinecone.init(api_key="***", environment="...")
             index = pinecone.Index("langchain-demo")
             embeddings = OpenAIEmbeddings()
-            vectorstore = Pinecone(index, embeddings.embed_query, "text")
+            vectorstore = Pinecone(index, embeddings, "text")
     """
 
     def __init__(
         self,
         index: Any,
-        embedding_function: Callable,
+        embedding: Union[Embeddings, Callable],
```

Comment:
@hwchase17 thoughts on this for backwards compat

Reply:
lgtm... its backward compatible right?

Reply:
afaict should be? unless someone was specifying it as a kwarg. can add a test to make sure